app/controllers/pets_controller.rb

```diff
@@ -1,7 +1,13 @@
 class PetsController < ApplicationController
 
   def index
-    @pets = Pet.all.order_by_adoption_status
+    if params[:adoption_status] == "true"
+      @pets = Pet.all.adoptable_only
+    elsif params[:adoption_status] == "false"
+      @pets = Pet.all.pending_only
+    else
+      @pets = Pet.all.order_by_adoption_status
+    end
```

Comment:
Looks at the params to decide which pets are shown. Again, I am not sure if this is too much logic for the controller.